### PR TITLE
[SYCL] Use exclusive queues for blocked commands.

### PR DESCRIPTION
**Pull request by romanovvlad**

### CODE REVIEW

File: sycl/test/lit.cfg [hunk 1]
```diff
@@ -98,3 +98,9 @@ path = config.environment['PATH']
 path = os.path.pathsep.join((config.llvm_tools_dir, path))
 config.environment['PATH'] = path
 
+# Set timeout for test = 10 mins
+try:
+    import psutil  # noqa: F401
+    lit_config.maxIndividualTestTime = 600
```
Comment (bader) [inline sycl/test/lit.cfg]: How did you get 10 minutes?
Is it enough for slowest configuration: e.g. debug+sanitizers+1thread+<my_slowest_cpu>+...?

File: sycl/test/sub_group/barrier.cpp [hunk 2]
```diff
@@ -44,16 +44,10 @@ template <typename T> void check(queue &Queue, size_t G = 240, size_t L = 60) {
...
-    // Temporary workaround to avoid waiting for the task while still blocked
-    // task is located in the same queue.
-    size_t sg_size = 0;
-    {
-      auto sgsizeacc = sgsizebuf.get_access<access::mode::read_write>();
-      sg_size = sgsizeacc[0];
-    }
-
     auto addacc = addbuf.template get_access<access::mode::read_write>();
+    auto sgsizeacc = sgsizebuf.get_access<access::mode::read_write>();
```
Comment (bader) [inline sycl/test/sub_group/barrier.cpp]: Please, add some information about the fix.
What we did to fix the problem?
If it's the commit from this PR, please, consider merging them together or reference the commit id.

File: sycl/test/lit.cfg (hunk 1 above)
Comment (romanovvlad) [reply sycl/test/lit.cfg]: Just took the same number as used for lldb testing. I think there is no problems with adjusting it if we encounter problems.
Note: the timeout is per LIT test, not for the whole suite.

File: sycl/test/sub_group/barrier.cpp (hunk 2 above)
Comment (romanovvlad) [reply sycl/test/sub_group/barrier.cpp]: I just revert changes I made earlier to work around problems that now resolved in  "[SYCL] Use exclusive queues for blocked commands. "

**Comment by romanovvlad** [comment]
> 
> 
> Please, move "[SYCL] Use exclusive queues for blocked commands" commit log message to the documentation/code comments.

Done.

File: sycl/test/sub_group/barrier.cpp (hunk 2 above)
Comment (bader) [reply sycl/test/sub_group/barrier.cpp]: I think it should be part of the "[SYCL] Use exclusive queues for blocked commands. " commit then.

Comment (romanovvlad) [reply sycl/test/sub_group/barrier.cpp]: Ok, merged.

**Pull request merged**